### ISSUE TITLE
No accuracy enhancement or loss reduction epochs

### ISSUE THREAD

**Issue by dannypfetzer** [rstudio/keras#634]
Hi everybody, my model doesn´t improve at all over the epochs and I can´t figure out why. 

First of all, this is my model:
`model <- keras_model_sequential() %>% 
  layer_dense(units = 16, activation = "relu", input_shape = c(1218)) %>% 
  layer_dense(units = 16, activation = "relu") %>%
  layer_dense(units = 1, activation = "sigmoid")

model %>% compile(
  optimizer = "rmsprop",
  loss = "binary_crossentropy",
  metrics = c("accuracy")
)

model %>% compile(
  optimizer = optimizer_rmsprop(lr = 0.001),
  loss = "binary_crossentropy",
  metrics = c("accuracy")
)

model %>% compile(
  optimizer = "rmsprop",
  loss = "binary_crossentropy",
  metrics = c("accuracy")
)

history <- model %>% fit(
  x_train,
  y_train,
  epochs          = 7,
  batch_size      = 500,
  validation_data = list(x_test, y_test)
)`

The Inputs look like this:
x_train
![grafik](https://user-images.githubusercontent.com/44265047/50998708-9093ef00-1528-11e9-8f94-d489bfdeb00a.png)
with this structure: 
 num [1:3500, 1:1218] 71567 73437 69071 71610 67758 ...
 - attr(*, "dimnames")=List of 2
  ..$ : chr [1:3500] "14" "15" "16" "17" ...
  ..$ : NULL
y_train
![grafik](https://user-images.githubusercontent.com/44265047/50998801-dfda1f80-1528-11e9-8f54-0e63f8fefd2d.png)
with this structure:
 num [1:3500] 0 0 0 0 0 0 0 0 0 0 ...

x_test and y_test have the same format and this is what I get from every epoch:
loss: 14.5759 - acc: 0.0857 - val_loss: 5.6937 - val_acc: 0.6429

These figures never change and I have absolutely no idea why. I hope someone can help me with this.

**Comment by skeydan** [comment]
Try scaling/normalizing the data.

**Comment by dannypfetzer** [comment]
@skeydan Thank you but even when I replace all the values in x_train with ones and zeros I get no improvement over the epochs.

**Comment by kevinykuo** [comment]
@dannypfetzer your `x_train` data doesn't look binary so you should try actually scaling them as @skeydan suggested. If that doesn't help try lowering the learning rate and maybe inspecting your gradients, e.g.
```
weights <- model$model$trainable_weights
gradients <- model$optimizer$get_gradients(model$total_loss, weights)
input_tensors <- list(model$input, model$sample_weights[[1]], model$targets[[1]])
get_gradients <- k_function(input = input_tensors, outputs = gradients)
inputs <- list(x_train, array(1:dim(x_train)[[1]]), y_train)
get_gradients(inputs)
```
